simapp/app.go

```diff
@@ -289,11 +290,19 @@ func NewSimApp(
 		AddRoute(distrtypes.RouterKey, distr.NewCommunityPoolSpendProposalHandler(app.DistrKeeper)).
 		AddRoute(upgradetypes.RouterKey, upgrade.NewSoftwareUpgradeProposalHandler(app.UpgradeKeeper)).
 		AddRoute(ibchost.RouterKey, ibcclient.NewClientUpdateProposalHandler(app.IBCKeeper.ClientKeeper))
+
+	// routes are added to tallyrouter after creation of govkeeper, so the govkeeper can be passed into tally routes
+	tallyRouter := govtypes.NewTallyRouter()
+
 	app.GovKeeper = govkeeper.NewKeeper(
 		appCodec, keys[govtypes.StoreKey], app.GetSubspace(govtypes.ModuleName), app.AccountKeeper, app.BankKeeper,
-		&stakingKeeper, govRouter,
+		&stakingKeeper, govRouter, tallyRouter,
 	)
 
+	// TODO: Make requirement to set a Root Tally Route, panic or something if not set
+	tallyRouter.AddRoute(govtypes.RootTallyRoute, stakingtally.NewStakingTallyHandler(app.GovKeeper, app.StakingKeeper))
```

Comment:
govtypes use NewStakingTallyHandler?

Reply:
This is for the "root" tallyroute.  It's basically the fallback tallystrategy if the requested one isn't registered on chain.  In app.go, by setting the "root" strategy to be stakingtally, it uses that as the fallback default.